src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs

```diff
@@ -0,0 +1,176 @@
+    {
+        private static readonly string _expectedResponseString = "Hello from delegatee";
+
+        [ConditionalFact]
+        [DelegateSupportedCondition(true)]
+        public async Task DelegateRequestTest()
+        {
+            var queueName = Guid.NewGuid().ToString();
+             using var receiver = Utilities.CreateHttpServer(out var receiverAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+            },
+            options =>
+            {
+                options.RequestQueueName = queueName;
+            });
+
+            DelegationRule wrapper = default;
+
+            using var delegator = Utilities.CreateHttpServer(out var delegatorAddress, httpContext =>
+            {
+                var transferFeature = httpContext.Features.Get<IHttpSysRequestTransferFeature>();
+                transferFeature.TransferRequest(wrapper);
+                return Task.FromResult(0);
+            });
+
+            var delegationProperty = delegator.Features.Get<IServerDelegationFeature>();
+            wrapper = delegationProperty.CreateDelegationRule(queueName, receiverAddress);
+
+            var responseString = await SendRequestAsync(delegatorAddress);
+            Assert.Equal(_expectedResponseString, responseString);
+            wrapper?.Dispose();
+        }
+
+        [ConditionalFact]
+        [DelegateSupportedCondition(true)]
+        public async Task DelegateAfterWriteToResponseBodyShouldThrowTest()
+        {
+            var queueName = Guid.NewGuid().ToString();
+            using var receiver = Utilities.CreateHttpServer(out var receiverAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+            },
+            options =>
+            {
+                options.RequestQueueName = queueName;
+            });
+
+            DelegationRule wrapper = default;
+
+            using var delegator = Utilities.CreateHttpServer(out var delegatorAddress, async httpContext =>
+            {
+                await httpContext.Response.WriteAsync(_expectedResponseString);
+                var transferFeature = httpContext.Features.Get<IHttpSysRequestTransferFeature>();
+                Assert.False(transferFeature.IsTransferable);
+                Assert.Throws<InvalidOperationException>(() => transferFeature.TransferRequest(wrapper));
+            });
+
+            var delegationProperty = delegator.Features.Get<IServerDelegationFeature>();
+            wrapper = delegationProperty.CreateDelegationRule(queueName, receiverAddress);
+
+            var responseString = await SendRequestAsync(delegatorAddress);
+            Assert.Equal(_expectedResponseString, responseString);
+            wrapper?.Dispose();
+        }
+
+        [ConditionalFact]
+        [DelegateSupportedCondition(true)]
+        public async Task WriteToBodyAfterDelegateShouldNoOp()
```

Comment:
You're not actually writing to the body anymore...

Reply:
Adding back the write